modules/src/clients/ics07_tendermint/client_state.rs

```diff
@@ -167,6 +167,41 @@ impl ClientState {
             clock_drift: self.max_clock_drift,
         })
     }
+
+    /// Verify the time and height delays
+    pub fn verify_delay_passed(
+        current_time: Timestamp,
+        current_height: Height,
+        processed_time: Timestamp,
+        processed_height: Height,
+        delay_period_time: Duration,
+        delay_period_blocks: u64,
+    ) -> Result<(), Error> {
+        let time = (processed_time + delay_period_time).map_err(Error::timestamp_overflow)?;
+        if !(current_time == time || current_time.after(&time)) {
+            return Err(Error::not_enough_time_elapsed());
+        }
+
+        if current_height < processed_height.add(delay_period_blocks) {
+            return Err(Error::not_enough_blocks_elapsed());
```

Comment:
Same comment as above, about augmenting the error type. Ideally, we'd also capture the heights in the error message:

```rust
            return Err(Error::not_enough_blocks_elapsed(current_height, processed_height, delay_period_blocks));
```

Reply:
5c1ee1d829aa598f0c649ef4534cf6de08d083ab